### ISSUE TITLE
Can Pyglet's Controller Input Handling Be Integrated with Other UI Frameworks?

### ISSUE THREAD

**Issue by carimatics** [pyglet/pyglet#997]
(I'm sorry for my poor English)

Hi,

I am currently developing software for automating Nintendo Switch games as a hobby project. As part of this software utility, I'd like to create a feature that receives input from a controller to operate Nintendo Switch games and generate automation code. During my research for suitable libraries, I came across the input.controller functionality of Pyglet and verified that it could handle inputs from the controller.

My issue is that Pyglet does not seem to have rich GUI capabilities for creating the utility interface (apologies if it does). I spent a day trying to code for UI construction but found it challenging and am not confident in my ability to maintain the code even if I could complete it. Therefore, I thought that my problem could be solved if I could combine Pyglet's features with another UI framework, like Flet.

If there is a way to do this, I would greatly appreciate your guidance. Alternatively, I would be very pleased to learn about any other possible solutions.

Of course, I am aware that there is no fault with Pyglet; my usage is just quite special.

Thanks.

**Comment by shenjackyuanjie** [comment]
hmmmm
you may want to try pyimgui[pyglet] backend, they support making ui easier

**Comment by carimatics** [comment]
Thank you so much for your prompt response!
I wasn't aware of that library.
I definitely want to give it a try.

**Comment by caffeinepills** [comment]
There is an example here of using pyglet with Qt.

https://github.com/pyglet/pyglet/blob/master/contrib/qt_sprite_preview.py

**Comment by carimatics** [comment]
Thank you for providing the example!
I will take it as a reference.

**Comment by carimatics** [comment]
After received the advices, I experimented with various library combinations and also sought solutions by asking questions in other communities.
As a result, I successfully achieved the desired behavior by combining [Flet](https://pypi.org/project/flet/) with [pyjoystick](https://pypi.org/project/pyjoystick/).
While a new probrem has emerged, I have been able to move forward from the problem initially raised in this issue, so I would like to close it now.
I am grateful for your advice.

Thank you very much.
